validation/validation_test.go

```diff
@@ -17,51 +22,90 @@ var (
 	runtime = "runc"
 )
 
+type validation struct {
+	test        func(string, string, *rspec.Spec) error
+	description string
+}
+
 func init() {
 	runtime = os.Getenv("RUNTIME")
 }
 
-func runtimeValidate(runtime string, g *generate.Generator) error {
+func TestValidateRuntimeInside(t *testing.T) {
+	g, err := getDefaultGenerator()
+	if err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+	g.SetProcessArgs([]string{"/runtimetest"})
+
+	if err := runtimeInsideValidate(runtime, g); err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+}
+
+func TestValidateRuntimeOutside(t *testing.T) {
+	g, err := getDefaultGenerator()
+	if err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+	g.SetProcessArgs([]string{"sleep", "50"})
+
+	if err := runtimeOutsideValidate(runtime, g); err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+}
+
+func runtimeInsideValidate(runtime string, g *generate.Generator) error {
 	// Find the runtime binary in the PATH
 	runtimePath, err := exec.LookPath(runtime)
 	if err != nil {
 		return err
 	}
 
-	// Setup a temporary test directory
-	tmpDir, err := ioutil.TempDir("", "ocitest")
+	bundleDir, rootfsDir, err := prepareBundle(g)
 	if err != nil {
 		return err
 	}
-	defer os.RemoveAll(tmpDir)
+	defer os.RemoveAll(bundleDir)
 
-	// Create bundle directory for the test container
-	bundleDir := tmpDir + "/busybox"
-	if err := os.MkdirAll(bundleDir, 0755); err != nil {
+	// Copy the runtimetest binary to the rootfs
+	err = fileutils.CopyFile("../runtimetest", filepath.Join(rootfsDir, "runtimetest"))
+	if err != nil {
 		return err
 	}
 
-	// Untar the root fs
-	untarCmd := exec.Command("tar", "-xf", "../rootfs.tar.gz", "-C", bundleDir)
-	output, err := untarCmd.CombinedOutput()
-	if err != nil {
-		fmt.Println(string(output))
+	// TODO: Use a library to split run into create/start
+	// Launch the OCI runtime
+	containerID := uuid.NewV4()
+	runtimeCmd := exec.Command(runtimePath, "run", containerID.String())
+	runtimeCmd.Dir = bundleDir
+	runtimeCmd.Stdin = os.Stdin
+	runtimeCmd.Stdout = os.Stdout
+	runtimeCmd.Stderr = os.Stderr
+	if err = runtimeCmd.Run(); err != nil {
 		return err
 	}
 
-	// Copy the runtimetest binary to the rootfs
-	err = fileutils.CopyFile("../runtimetest", filepath.Join(bundleDir, "runtimetest"))
+	return nil
+}
 
-	// Generate test configuration
-	err = g.SaveToFile(filepath.Join(bundleDir, "config.json"), generate.ExportOptions{})
+func runtimeOutsideValidate(runtime string, g *generate.Generator) error {
+	// Find the runtime binary in the PATH
+	runtimePath, err := exec.LookPath(runtime)
+	if err != nil {
+		return err
+	}
+
+	bundleDir, _, err := prepareBundle(g)
 	if err != nil {
 		return err
 	}
+	//	defer os.RemoveAll(bundleDir)
 
 	// TODO: Use a library to split run into create/start
 	// Launch the OCI runtime
```

Comment:
You're using `create` below, so this comment is stale here.  You will want to run `kill` and `delete` to clean up though.